src/NBitcoin/ChainedHeader.cs

```diff
@@ -4,6 +4,63 @@
 
 namespace NBitcoin
 {
+    /// <summary>
+    /// Represents the state of the availability of a block.
+    /// </summary>
+    public enum BlockDataAvailabilityState
+    {
+        /// <summary>
+        /// A <see cref="BlockHeader"/> is presented, the block data is not available.
+        /// </summary>
+        HeaderOnly,
+
+        /// <summary>
+        /// We are interested in downloading the <see cref="Block"/> that is being represented by the current <see cref="BlockHeader"/>. 
+        /// This happens when a header is a part of the chain that can potentially replace our consensus tip 
+        /// because its tip's total chain work is greater comparing to what we currently have.
+        /// </summary>
+        BlockRequired,
+
+        /// <summary>
+        /// The <see cref="Block"/> was downloaded and is available, but it may not be reachable directly but via a store.
+        /// </summary>
+        BlockAvailable
+    }
+
+    /// <summary>
+    /// Represents the validation level of a block.
+    /// </summary>
+    public enum ValidationState
+    {
+        /// <summary>
+        /// No validation was performed.
+        /// </summary>
+        Unknown,
+
+        /// <summary>
+        /// We have a valid block header.
+        /// </summary>
+        HeaderValidated,
+
+        /// <summary>
+        /// Minimum validation required because the block is assumed to be valid.
+        /// This state is used for blocks before the last checkpoint or for blocks that are on the chain of assume valid block.
```

Comment:
=> `... for block headers before the last checkpoint, which are on the correct chain, or for block headers that are ...`